org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/engine/connected/ConnectedEngineFacade.java

```diff
@@ -647,17 +601,37 @@ public Optional<ServerHotspot> getServerHotspot(String hotspotKey, String projec
 
   @Override
   public ProjectBranches getServerBranches(String projectKey) {
-    return withEngine(engine -> engine.getServerBranches(projectKey)).orElse(new ProjectBranches(Set.of(), Optional.empty()));
+    return withEngine(engine -> engine.getServerBranches(projectKey)).orElse(new ProjectBranches(Set.of("master"), "master"));
   }
 
   @Override
   public void sync(Set<String> projectKeysToUpdate, IProgressMonitor monitor) {
     doWithEngine(engine -> {
       engine.sync(createEndpointParams(), buildClientWithProxyAndCredentials(), projectKeysToUpdate,
         new WrappedProgressMonitor(monitor, "Synchronize projects storage for connection '" + getId() + "'"));
+      // Force recompute of best server branch
+      VcsService.clearVcsCache();
+
+      updateAllProjectIssuesForCurrentBranch(engine, projectKeysToUpdate, monitor);
     });
-    // Force recompute of best server branch
-    VcsService.clearVcsCache();
+  }
+
+  private void updateAllProjectIssuesForCurrentBranch(ConnectedSonarLintEngine engine, Set<String> projectKeys, IProgressMonitor monitor) {
+    for (var projectKey : projectKeys) {
+      var boundProjects = getBoundProjects(projectKey);
+
+      Set<String> activeBranches = new HashSet<>();
+      for (var project : boundProjects) {
+        activeBranches.add(VcsService.getServerBranch(project));
+      }
+
+      activeBranches.forEach(b -> {
+        engine.downloadAllServerIssues(createEndpointParams(), buildClientWithProxyAndCredentials(), projectKey, b,
```

Comment:
Shouldn't we call `syncServerIssues` here instead ? And also `syncServerTaintIssues` ?